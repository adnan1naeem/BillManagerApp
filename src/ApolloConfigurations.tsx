import { ApolloClient } from 'apollo-client'
import { HttpLink } from 'apollo-link-http'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { WebSocketLink } from 'apollo-link-ws';
import { split } from 'apollo-link';
import { getMainDefinition } from 'apollo-utilities';
import fetch from 'node-fetch';


// const wsLink = new WebSocketLink({
//   uri: 'wss://subscriptions.graph.cool/v1/cjk3wyu0c5you0107e5a48dgk',
//   options: {
//     reconnect: true
//   }
// });


``

const httpLink = new HttpLink({
  uri: 'https://api.graph.cool/simple/v1/cjk3wyu0c5you0107e5a48dgk',
  fetch
});


// const link = split(
//   // split based on operation type
//   ({ query }) => {
//     const { kind, operation }: any = getMainDefinition(query);
//     return kind === 'OperationDefinition' && operation === 'subscription';
//   },
//   wsLink,
//   httpLink,
// );


const client = new ApolloClient({
  ssrMode: true,
  link: httpLink,
  cache: new InMemoryCache(),
})




export { client };