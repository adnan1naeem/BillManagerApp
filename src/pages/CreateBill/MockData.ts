const AssetList = [
  { label: 'AC', value: 'AC' },
  { label: 'BULB', value: 'BULB' },
  { label: 'FAN', value: 'FAN' },
];
const SiteList = [
  { label: 'Home', value: 'Home' },
  { label: 'factory', value: 'factory' },
  { label: 'shop', value: 'shop' }
];

const DateList = [
  { value: 'January', label: 'January' },
  { value: 'February', label: 'February' },
  { value: 'March', label: 'March' },
  { value: 'April', label: 'April' },
  { value: 'May', label: 'May' },
  { value: 'June', label: 'June' },
  { value: 'July', label: 'July' },
  { value: 'August', label: 'August' },
  { value: 'September', label: 'September' },
  { value: 'October', label: 'October' },
  { value: 'November', label: 'November' },
  { value: 'December', label: 'December' }
];


export {AssetList,SiteList,DateList}