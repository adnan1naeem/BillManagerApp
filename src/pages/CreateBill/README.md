#  BillListing Component

## Compatibility

| 🌏 Web                   | 🖥 Electron              | 📱 React Native    |
| :----------------------: | :----------------------: | :----------------: |
| :heavy_multiplication_x: | :heavy_multiplication_x: | :heavy_check_mark: |

## Screenshots

| 🌏 Web | 🖥 Electron | 📱 React Native                            |
| :----: | :---------: | :----------------------------------------: |
| TBD    | TBD         | ![CreateBill](./screenshot/CreateBill.png) |




## Props

| Name   | Type   | Default | Description                           |
| :----- | :----- | :------ | :------------------------------------ |
| mutate | object |         | It is used for mutation of Data       |
| Name   | string |         | It is used for adding Name  of Bill   |
| month  | string |         | It is used for adding month  of Bill  |
| budget | string |         | It is used for adding budget  of Bill |


## How to use

```react
import React from 'react';
        <CreateBill  name="ADNAN"  month="January" budget={2000}/>
```

## Tests
