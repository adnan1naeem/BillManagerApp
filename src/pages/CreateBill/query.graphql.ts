

import gql from 'graphql-tag'

export const Mutationquery = gql`

mutation($name:String!,$budget:String!,$unitrate:String!,$site:String!,$asset:String!,$month:String!){
  createBill(
    name:$name,
    asset:$asset,
    site:$site,
    month:$month,
    unitrate:$unitrate
    budget:$budget
  )
  {
    id
    asset
    budget
  }
}
`


