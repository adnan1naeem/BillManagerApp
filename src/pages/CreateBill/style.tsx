export const styles: any = {

  Container: {
    display: 'flex',
    flexDirection: 'column',
    padding: 5
  },

  error: {
    color: 'red'
  }
}