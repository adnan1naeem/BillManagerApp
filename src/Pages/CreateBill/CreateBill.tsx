import React from 'react'
import ToolbarAndroid from "../../components/AppBar/AppBar.component"
import { ScrollView, StatusBar, ActivityIndicator } from "react-native";
import FormInput from "../../components/FormInput/FormInput.component"
import FormPicker from "../../components/FormPicker/FormPicker.component"
import InputLabel from "../../components/InputLabel/InputLabel.component"
import { Mutation, graphql } from "react-apollo";
import { Button } from "react-native"
import { Mutationquery } from "./query.graphql"
import { withRouter } from 'react-router-native';
import { query } from "../BillListing/query.graphql"
import { Formik } from 'formik';
import { View } from 'react-native';
import { styles } from "./style"
import { AssetList, DateList, SiteList } from "./MockData";
export interface RouterProps {
	push: (route: string) => void
}


export interface MutationProps {


  /**
	 * props for variables of mutation 
	 *  
	 */
	variables: any,
}
export interface Props {


  /**
	 * props of mutation of data
	 * 
	 */
	mutate: any,


  /**
	 *  history props for routing 
	 * 
	 */
	history: RouterProps,


  /**
	 *  loading props  check for data posting 
	 * 
	 */

	loading: boolean,

	/**
	 *  Name props  check for adding BillName
	 * 
	 */

	Name: string,



	/**
	 *  Name props  check for adding SiteName for Bill
	 * 
	 */

	Site: string,

	/**
	 *  Name props  check for adding AssetName for Bill
	 * 
	 */

	Asset: string,


	/**
	 *  Name props  check for adding AssetName for Bill
	 * 
	 */

	month: string,

	/**
	 *  budget props  check for adding budget for Bill
	 * 
	 */

	budget: string,
	/**
	 *  unitRate props  check for adding unitRate for Bill
	 * 
	 */


	unitRate: string

}
const CreatBill = (props: Props) => {

	return (
		<ScrollView>

			<StatusBar barStyle="dark-content" hidden={false} />

			<ToolbarAndroid
				title="Create"
				BackButton={true}
			/>

			<Formik
				initialValues={{ Name: props.Name, Site: props.Site, Asset: props.Asset, month: props.month, budget: props.budget, unitRate: props.unitRate }}
				onSubmit={values =>

					props.mutate({
						variables: {
							name: values.Name,
							budget: values.budget,
							site: values.Site,
							asset: values.Asset,
							unitrate: values.unitRate,
							month: values.month
						},
						refetchQueries: [
							{ query: query }
						]

					}).then(() => {

						if (props.loading == true) {
							<ActivityIndicator />
						}
						props.history.push('/listing');
					})

				}>

				{({ handleChange, handleSubmit, values }) => (

					<View style={styles.Container}>
						<FormInput IconColor="green" IconSize={22} IconName="md-checkmark-circle" value={values.Name} label="BillName" onValueChange={handleChange('Name')} />
						<FormPicker IconSize={22} IconName="md-checkmark-circle" IconColor="green" handleChange={handleChange('Asset')} label="Select Device" Items={AssetList} value={values.Asset} />
						<FormPicker IconSize={22} IconName="md-checkmark-circle" IconColor="green" value={values.month} label="Select Month" handleChange={handleChange('month')} Items={DateList} />
						<InputLabel keyboardType='numeric' value={values.unitRate} onChangeText={handleChange('unitRate')} label="Unit Rate" />
						<InputLabel keyboardType='numeric' value={values.budget} onChangeText={handleChange('budget')} label="Budget" />
						<Button onPress={handleSubmit} title="Save" />
					</View>
				)}
			</Formik>

		</ScrollView>

	);
}

export default graphql(Mutationquery)(withRouter(CreatBill));
