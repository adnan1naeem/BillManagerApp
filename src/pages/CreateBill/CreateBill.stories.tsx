import React from 'react';
import { storiesOf } from '@storybook/react-native';
import CreateBill from './CreateBill';
import { withKnobs, text, } from '@storybook/addon-knobs/react';
import { View } from 'react-native';


storiesOf('CreateBill', module)
    .addDecorator(getStory => <View>{getStory()}</View>)
    .addDecorator(withKnobs)
    .add('Page', () => (
        <CreateBill />
    ))


