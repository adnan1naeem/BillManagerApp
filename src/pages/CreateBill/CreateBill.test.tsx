import * as React from 'react'
import { shallow, configure, mount } from 'enzyme';
import CreateBill from "./CreateBill";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import { ApolloProvider } from 'react-apollo'
import { ApolloClient } from 'apollo-client'
import { HttpLink } from 'apollo-link-http'
import { InMemoryCache } from 'apollo-cache-inmemory'
import { NativeRouter, BackButton, Route } from 'react-router-native';
import { client } from "../../ApolloConfigurations"
configure({ adapter: new Adapter() })



test('renders correctly', () => {
  shallow(<ApolloProvider client={client}><NativeRouter><CreateBill /></NativeRouter></ApolloProvider>);
});


test('renders correctly', () => {
  const tree = renderer.create(
    <ApolloProvider client={client}><NativeRouter><CreateBill /></NativeRouter></ApolloProvider>).toJSON();
  expect(tree).toMatchSnapshot();
});



