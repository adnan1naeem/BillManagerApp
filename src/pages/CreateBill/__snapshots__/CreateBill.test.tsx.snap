// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "blue",
          "display": "flex",
          "flex": 1,
          "flexDirection": "row",
          "height": 56,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
          }
        }
      >
        Create
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "padding": 5,
        }
      }
    >
      <View
        styles={
          Object {
            "display": "flex",
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "marginLeft": 20,
            }
          }
        >
          BillName
        </Text>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <TextInput
            IconColor="green"
            IconName="md-checkmark-circle"
            IconSize={22}
            allowFontScaling={true}
            label="BillName"
            onChangeText={[Function]}
            onValueChange={[Function]}
            style={
              Object {
                "borderBottomColor": "#4381FF",
                "color": "#4381FF",
                "flex": 1,
                "height": 30,
                "width": 300,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "paddingTop": 20,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "marginLeft": 20,
              "marginTop": 5,
            }
          }
        >
          Select Device
        </Text>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <View
            style={
              Object {
                "borderBottomColor": "#4381FF",
                "flex": 1,
              }
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "AC",
                    "textColor": undefined,
                    "value": "AC",
                  },
                  Object {
                    "label": "BULB",
                    "textColor": undefined,
                    "value": "BULB",
                  },
                  Object {
                    "label": "FAN",
                    "textColor": undefined,
                    "value": "FAN",
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
            "paddingTop": 20,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "marginLeft": 20,
              "marginTop": 5,
            }
          }
        >
          Select Month
        </Text>
        <View
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          />
          <View
            style={
              Object {
                "borderBottomColor": "#4381FF",
                "flex": 1,
              }
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "January",
                    "textColor": undefined,
                    "value": "January",
                  },
                  Object {
                    "label": "February",
                    "textColor": undefined,
                    "value": "February",
                  },
                  Object {
                    "label": "March",
                    "textColor": undefined,
                    "value": "March",
                  },
                  Object {
                    "label": "April",
                    "textColor": undefined,
                    "value": "April",
                  },
                  Object {
                    "label": "May",
                    "textColor": undefined,
                    "value": "May",
                  },
                  Object {
                    "label": "June",
                    "textColor": undefined,
                    "value": "June",
                  },
                  Object {
                    "label": "July",
                    "textColor": undefined,
                    "value": "July",
                  },
                  Object {
                    "label": "August",
                    "textColor": undefined,
                    "value": "August",
                  },
                  Object {
                    "label": "September",
                    "textColor": undefined,
                    "value": "September",
                  },
                  Object {
                    "label": "October",
                    "textColor": undefined,
                    "value": "October",
                  },
                  Object {
                    "label": "November",
                    "textColor": undefined,
                    "value": "November",
                  },
                  Object {
                    "label": "December",
                    "textColor": undefined,
                    "value": "December",
                  },
                ]
              }
              onChange={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={Object {}}
        >
          Unit Rate
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          label="Unit Rate"
          onChangeText={[Function]}
          style={
            Object {
              "borderBottomColor": "#4381FF",
              "color": "#4381FF",
              "height": 30,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={Object {}}
        >
          Budget
        </Text>
        <TextInput
          allowFontScaling={true}
          keyboardType="numeric"
          label="Budget"
          onChangeText={[Function]}
          style={
            Object {
              "borderBottomColor": "#4381FF",
              "color": "#4381FF",
              "height": 30,
            }
          }
        />
      </View>
      <View
        accessibilityComponentType="button"
        accessibilityTraits={
          Array [
            "button",
          ]
        }
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Save
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
